lms/models/grouping.py

```diff
@@ -0,0 +1,73 @@
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+from lms.db import BASE
+from lms.models import TimestampedModelMixin
+from lms.models.application_settings import ApplicationSettings
+
+MAX_GROUP_NAME_LENGTH = 25
+
+
+class Grouping(TimestampedModelMixin, BASE):
+    __tablename__ = "grouping"
+    __mapper_args__ = {"polymorphic_identity": "grouping", "polymorphic_on": "type"}
+    __table_args__ = (
+        sa.UniqueConstraint("application_instance_id", "authority_provided_id"),
+    )
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    application_instance_id = sa.Column(
+        sa.Integer(),
+        sa.ForeignKey("application_instances.id", ondelete="cascade"),
+        nullable=False,
+    )
+    application_instance = sa.orm.relationship("ApplicationInstance")
+
+    authority_provided_id = sa.Column(sa.UnicodeText(), nullable=False)
+
+    parent_id = sa.Column(
+        sa.Integer(),
+        sa.ForeignKey("grouping.id", ondelete="cascade"),
+        nullable=True,
+    )
+
+    #: ID on the LMS, not unique acrooss LMS or even in the same LMS instance
+    lms_id = sa.Column(sa.String(), nullable=True)
+
+    #: Full name given on the LMS (e.g. "A course name 101")
+    lms_name = sa.Column(sa.UnicodeText(), nullable=False)
```

Comment:
Is this gonna come from the `context_title` launch param? If so I wonder if we should call it `context_title` or maybe just `title`?

Reply:
Note that this not a `Course` table, its the grouping one with common fields to all of them, so I think `lms_name` works better to store a course name, a section name and a group name